### PR TITLE
feat: Added unstack function to paddle tensor class

### LINKED ISSUE #23498: unstack

### PR DESCRIPTION
**Pull request by arshPratap**
# PR Description
Adds unstack function to paddle tensor class

## Related Issue

Close #23498 

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by arshPratap** [comment]
Hi @bipinKrishnan ... no `failing_tests` here..could you take a look ?

**Comment by bipinKrishnan** [comment]
Hey @arshPratap, could you please fix the conflicts and push again?

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 1]
```diff
@@ -765,3 +765,9 @@ def remainder(self, y, name=None):
 
     def is_floating_point(self):
         return paddle_frontend.is_floating_point(self._ivy_array)
+
+    @with_supported_dtypes(
+        {"2.5.1 and below": ("float32", "float64", "int32", "int64")}, "paddle"
+    )
+    def unstack(self, axis=0, name=None):
```
Comment (bipinKrishnan) [inline ivy/functional/frontends/paddle/tensor/tensor.py:772]: The function takes another argument `num`. You could refer to the docs [here](https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/unstack_en.html).

Comment (arshPratap) [reply ivy/functional/frontends/paddle/tensor/tensor.py]: @bipinKrishnan Done

**Comment by arshPratap** [comment]
> Member

Thanks for the review @bipinKrishnan ..not sure I was running the tests locally and I found that all the unstack function tests irrespective of the frontend (tensorflow/paddle) are failing..and I am pretty sure its not related to the changes made in this PR

**Comment by arshPratap** [comment]
> > Hey @arshPratap, could you please fix the conflicts and push again?
> 
> Thanks for the review @bipinKrishnan ..not sure but when I was running the tests locally and I found that all the unstack function tests irrespective of the frontend (tensorflow/paddle) are failing..and I am pretty sure its not related to the changes made in this PR

Hi @bipinKrishnan  .. I believe there is an issue while running the unstack function tests and to override these failures I had to set the `generate_frontend_arrays` flag to `False` .. I am not sure if this is the right way..could you take a look

**Comment by arshPratap** [comment]
Hi @bipinKrishnan  looks like the above errors have been fixed now..The tests are passing for the method and I have rebased the branch to the latest changes..
Could you take a look again ?
Thanks and sorry for trouble

**Comment by arshPratap** [comment]
@bipinKrishnan Could you take a look ?

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**